### ISSUE TITLE
How to choose distributed training mode

### ISSUE THREAD

**Issue by sneaxiy** [sql-machine-learning/sqlflow#2288]
As far as I know, TensorFlow models only support PS-Worker mode and XGBoost models only support allreduce mode to train. Can we support nccl/mpi allreduce mode for TensorFlow? Allreduce may be quite important especially in GPU training.
